src/s1reader/s1_burst_slc.py

```diff
@@ -468,59 +492,66 @@ def geometrical_and_steering_doppler(self, xstep=500, ystep=50):
         with azimuth time.
         Parameters
         ----------
-        xstep: int
-            Spacing along x direction [pixels]
-        ystep: int
-            Spacing along y direction [pixels]
+        range_step: int
+            Spacing along x/range direction [meters]
+        az_step: int
+            Spacing along y/azimuth direction [seconds]
 
         Returns
         -------
-        x : int
-           The index of samples in range direction as an 1D array
-        y : int
-           The index of samples in azimuth direction as an 1D array
-        total_doppler : float
-           Total Doppler which is the sum of the geometrical Doppler and
-           beam steering induced Doppler [Hz] as a 2D array
+           LUT2D object of total doppler in Hz as a function of the azimuth
+           time and slant range, or range and azimuth indices.
+           This correction needs to be added to the SLC tagged azimuth time to
+           get the corrected azimuth times.
         """
+        range_arange, az_arange = self._steps_to_aranges(range_step, az_step)
 
-        x = np.arange(0, self.width, xstep, dtype=int)
-        y = np.arange(0, self.length, ystep, dtype=int)
-        x_mesh, y_mesh = np.meshgrid(x, y)
+        # convert from meters to pixels
+        x_arange = (range_arange - self.starting_range) / self.range_pixel_spacing
+
+        # convert from seconds to pixels
+        rdrgrid = self.as_isce3_radargrid()
+        y_arange = (az_arange - rdrgrid.sensing_start) / self.azimuth_time_interval
+
+        # compute az carrier components with pixels
+        x_mesh, y_mesh = np.meshgrid(x_arange, y_arange)
         az_carr_comp = self.az_carrier_components(
                                         offset=0.0,
                                         position=(y_mesh, x_mesh))
 
-        slant_range = self.starting_range + x * self.range_pixel_spacing
-        geometrical_doppler = self.doppler.poly1d.eval(slant_range)
+        geometrical_doppler = self.doppler.poly1d.eval(range_arange)
 
         total_doppler = az_carr_comp.antenna_steering_doppler + geometrical_doppler
 
-        return x, y, total_doppler
+        return isce3.core.LUT2d(range_arange, az_arange, total_doppler)
 
-    def doppler_induced_range_shift(self, xstep=500, ystep=50):
+    def doppler_induced_range_shift(self, range_step=500, az_step=50,
+                                    use_radar_coords=True):
```

Comment:
This parameter isn't used- where would it be passed to? Or should it be deleted?

Reply:
Apologies for more slop. Thank you for catching this.